### DISCUSSION TITLE
Plugin or custom fields for storage overview per VM

### DISCUSSION THREAD

**Discussion by MalfuncEddie** [netbox-community/netbox#9544] [Q&A] [answered]
Hi,

I''ve got a sync that populates netbox with vmware information. (vcpu memory, diskspace ...).
We also have some physical servers that I entered manualy. (and have SAN disks)
some vm's have a non datastore SAN disk. 

Problem:
I would like to have storage information in netbox. 
Eg: Per vm or physical servers I would like to add a "storage" device with certain specs (size)

I see an easy way and a hard way:

I could create a device and role for each lun and with a custom field link them to an vm or host.

hard way: 
I create a dedicated storage plugin

I do not have any experience writing plugins but what would be the best way forward?

**Comment by jeremystretch** [comment, marked as answer]
There's some discussion around this in FR #8356.
